pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -135,20 +138,56 @@ func (c *Controller) clusterBackupSync(ctx context.Context, interval time.Durati
 	for range ticker.Context(ctx, interval) {
 		clusters, err := c.clusterLister.List("", labels.NewSelector())
 		if err != nil {
-			logrus.Error(fmt.Errorf("[etcd-backup] error while listing clusters: %v", err))
+			log.Error(fmt.Errorf("[etcd-backup] error while listing clusters: %v", err))
 			return err
 		}
 		for _, cluster := range clusters {
-			logrus.Debugf("[etcd-backup] checking backups for cluster [%s]", cluster.Name)
-			if err := c.doClusterBackupSync(cluster); err != nil && !apierrors.IsConflict(err) {
-				logrus.Error(fmt.Errorf("[etcd-backup] error while syncing cluster backups for cluster [%s]: %v", cluster.Name, err))
+			log.Debugf("[etcd-backup] checking backups for cluster [%s]", cluster.Name)
+			if err = c.runWaitingBackups(cluster); err != nil {
+				log.Error(fmt.Errorf("[etcd-backup] error running waiting cluster backups for cluster [%s]: %v", cluster.Name, err))
 			}
+			if err = c.doRecurringBackup(cluster); err != nil {
+				log.Error(fmt.Errorf("[etcd-backup] error while syncing cluster backups for for cluster [%s]: %v", cluster.Name, err))
+			}
+		}
+	}
+	return nil
+}
+
+func (c *Controller) runWaitingBackups(cluster *v3.Cluster) error {
+	if cluster == nil || cluster.DeletionTimestamp != nil {
+		return nil
+	}
+	// check if the cluster is eligible for backup.
+	if !shouldBackup(cluster) {
+		return nil
+	}
+
+	backups, err := c.getBackupsList(cluster)
+	if err != nil {
+		return err
+	}
+
+	if anyBackupsRunning(cluster, backups) {
+		return nil
+	}
+
+	var next *v3.EtcdBackup
+	if next = nextBackup(backups); next != nil {
+		log.Infof("[etcd-backup] cluster [%s] backup starting from queue: %s", cluster.Name, next.Name)
+		if _, err := c.doBackupForCluster(next, cluster); err != nil {
+			return err
+		}
+		// rotate old backups on successful recurring backup
+		if !next.Spec.Manual {
+			return c.rotateExpiredBackups(cluster, getRecurringBackupsList(backups))
 		}
 	}
+
 	return nil
 }
 
-func (c *Controller) doClusterBackupSync(cluster *v3.Cluster) error {
+func (c *Controller) doRecurringBackup(cluster *v3.Cluster) error {
```

Comment:
Something that's bugging me here is that `doRecurringBackup` doesn't describe _what_ this function does. I think it would improve readability to be more specific, maybe call the function `createRecurringBackup` or `checkRecurringBackup.`